internal/pkg/cli/svc_override.go

```diff
@@ -13,32 +14,49 @@ import (
 	"github.com/aws/copilot-cli/internal/pkg/aws/identity"
 	"github.com/aws/copilot-cli/internal/pkg/aws/sessions"
 	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/template"
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+	"github.com/aws/copilot-cli/internal/pkg/term/selector"
 	"github.com/aws/copilot-cli/internal/pkg/workspace"
 	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
 )
 
 const (
 	// IaC options for overrides.
-	cdkIacToolkit = "cdk"
+	cdkIaCTool = "cdk"
 
 	// IaC toolkit configuration.
 	typescriptCDKLang = "typescript"
 )
 
-var iacToolkits = []string{
-	cdkIacToolkit,
+var validIaCTools = []string{
+	cdkIaCTool,
 }
 
 var validCDKLangs = []string{
 	typescriptCDKLang,
 }
 
+type stringWriteCloser interface {
+	fmt.Stringer
+	io.WriteCloser
+}
+
+type closableStringBuilder struct {
+	*strings.Builder
+}
+
+// Close implements the io.Closer interface for a strings.Builder and is a no-op.
+func (sb *closableStringBuilder) Close() error {
+	return nil
+}
+
 type overrideVars struct {
-	name    string
-	envName string
```

Comment:
Yeah, I guess this is related to your other comment :) 
I wanted to provide a simpler experience if possible with less prompts. The goal is to show _some_ useful resources so that the user can get started with writing CDK code.

If for some reason the resource isn't available in other environments, the `copilot deploy` or the `Build` stage in the pipeline will fail 👍 so the user should get an early signal that they have to update the CDK code with an if-statement

Reply:
hmm, I'm not sure I agree that a command completing successfully locally with one fewer prompt, then failing in the build stage, is a better experience than having one more prompt for the command and reducing the chance of build failure. I think the extra choice here is better than the confusion the failure would create 🤔